libs/utils/utils.js

```diff
@@ -861,9 +861,24 @@ export function scrollToHashedElement(hash) {
   });
 }
 
-export async function loadDeferred(area, blocks, config) {
+export function setupDeferredPromise() {
+  let resolveFn;
+  window.milo.deferredPromise = new Promise((resolve) => {
+    resolveFn = resolve;
+  });
+  return resolveFn;
+}
+
+export async function loadDeferred(area, blocks, config, resolveDeferred) {
   const event = new Event(MILO_EVENTS.DEFERRED);
   area.dispatchEvent(event);
+
+  if (area !== document) {
+    return;
+  }
+
+  resolveDeferred(true);
```

Comment:
Do we need this differentiation between event and Promise based on whether the area is a document? Could we not call `loadDeferred` inside the `loadArea` method only if the `isDoc` condition is met?

Reply:
[Tabs](https://github.com/adobecom/milo/blob/main/libs/blocks/tabs/tabs.js#L95) follows a similar pattern and uses this event too.